### ISSUE TITLE
[Crash] - when starting Oracle Virtualbox

### ISSUE THREAD

**Issue by sebbuku** [psifidotos/Latte-Dock#258]
Happens almost always when i start Virtualbox.
No difference if I start it from krunner, a launcher or latte-dock itself. 
Error log is attached.

Also I noticed when I select "always show a launcher when is not running" the wrong launcher is added, clicking it only throws the error: "Access denied to /usr/lib/virtualbox/VirtualBox."

[latte-dock-20170314-104916.kcrash.txt](https://github.com/psifidotos/Latte-Dock/files/840838/latte-dock-20170314-104916.kcrash.txt)

**Comment by psifidotos** [comment]
@sebbuku the backtrace shows that this may happen also with default plasma taskmanager...

Close Latte, use the default plasma taskmanager and confirm that isnt crashing

**Comment by psifidotos** [comment]
for crash reports please use also the #147 way

**Comment by sebbuku** [comment]
The launcher not being added properly I can confirm is the case with the standard task manager. 
But the crash only happens with latte.

**Comment by sebbuku** [comment]
DDD Backtrace:

![image](https://cloud.githubusercontent.com/assets/7261864/23895705/f62b32d2-08a6-11e7-954f-11ba43de0b3d.png)

**Comment by psifidotos** [comment]
Have you touched virtualbox desktop file in any way? this could explain both the launcher issue and the crash... the bug is inside libtaskmanager model but as long you dont have that problem with other launchers, it may be fixed with the following desktop file...

I am using openSuse Leap and works just fine...
opensuse's desktop file in use is

**Comment by sebbuku** [comment]
No I haven't. I tried running the attached desktop file, while Virtualbox start latte still crashes.

**Comment by psifidotos** [comment]
You could try setting no animations and not make it a Launcher to see if that helps...

unfortunately I can not report a crash to plasma libtaskmanager that I can not confirm or reproduce in order to help them trace that down....

**Comment by psifidotos** [comment]
@sebbuku you could try branch **fix258** in which i have removed some of our code in case we can do something in our regard... can you check please?

**Comment by sebbuku** [comment]
Just checked branch fix258. Now it works like a charm. I tried launching it in every way possible and no crashes.

**Comment by sebbuku** [comment]
still working

**Comment by psifidotos** [comment]
@sebbuku one more time please.... I am restoring the code parts piece by piece in order to find exactly the bug...

**Comment by sebbuku** [comment]
Pulled the latest changes and tested. Still no crash. I am happy to help you debug by the way.

**Comment by sebbuku** [comment]
And to make everything even weirder: I restarted my computer and when starting VirtualBox I had one latte crash, but only once. By every subsequent start I haven't got any additional crashes.

**Comment by psifidotos** [comment]
one more time please...

**Comment by psifidotos** [comment]
I just recreated fix258 branch and added some minor improvements... let's see if it helps...

**Comment by sebbuku** [comment]
Switched to branch fix258b and installed. No crashes.

**Comment by psifidotos** [comment]
@sebbuku sorry the fix258b was for another issue... I would like you to test fix258... what happens in that case ?

**Comment by sebbuku** [comment]
ok tried it now with the right branch. Still no crashes. :+1:

**Comment by psifidotos** [comment]
@sebbuku nice!!! this branch can be merged!!! ;)

**Comment by sebbuku** [comment]
Yay! Great job!

**Comment by baskinsy** [comment]
I'm experiencing the same issue with 0.6.2 on Opensuse Tumbleweed, is the fix including on this version?

**Comment by psifidotos** [comment]
it should... you could try master branch if you are using plasma 5.9 and later to inform us if this is still happening for you

**Comment by baskinsy** [comment]
I'm using 0.6.2 packaged for my distro and yes on plasma 5.9.x and the issue is always reproducible for me. I'll try to upload any debug information later as soon as I am on the specific system.

**Comment by psifidotos** [comment]
To produce a good backtrace you can follow #147 and you should at least install the debug packages for, latte, plasma-workspace and plasma desktop

**Comment by psifidotos** [comment]
You get a crash when you are executing  Virtual Box launcher or general when virtual box starts?

**Comment by baskinsy** [comment]
I'm experiencing the issue always with VirtualBox launcher yes. I'll test if I fire a VM directly and also try to debug and update here.

**Comment by baskinsy** [comment]
I'm unable to reproduce it today... Yesterday I had a huge update due to Opensuse Tumbleweed changed to gcc7 which fetched and installed almost all of the packages in my system.... Today latte is also working without crashing on executing the Virtualbox launcher..... Sorry for the false alert, the issue was reproducible always for me yesterday. In any case I experience it again, I'll update.

**Comment by psifidotos** [comment]
No worries... I am also using Tumbleweed so I might be able to reproduce it somehow
